File: python/n504_ConvertBase7.py
# https://leetcode-cn.com/problems/base-7/
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num < 0: return ("-" + self.convertToBase7(-num))
        if num < 7: return str(num)
        return self.convertToBase7(num // 7) + str(num % 7)
        
        
if __name__ == "__main__":
    solu = Solution()
    print(solu.convertToBase7(100))

File: cpp/n504_ConvertBase7.cpp
// https://leetcode-cn.com/problems/base-7/
#include <iostream>
using namespace std;

class Solution {
public:
    string convertToBase7(int num) {
        if (num < 0) return ("-" + convertToBase7(-num));
        if (num < 7) return to_string(num);
        return convertToBase7(num / 7) + to_string(num % 7);
    }
};

int main(){
    return 0;
}